###################################################################################################
Date: Mon, 17 Dec 2018 15:30:01 +0000
From 76ec600eb3ca9e3d59d5273de0904f9535be9882 Mon Sep 17 00:00:00 2001
From: Ben Gray <ben.gray@sky.uk>
Subject: [PATCH] This patch changes the uhid part of the HOG driver so that
 the bdaddr of the remote device is stored in the physical address field.
 Prior to this the bdaddr of the controller was set in the phys field and the
 remote bdaddr in the uniq field.

The previous behaviour is probably better, however it has two problems:
 1) all the current Sky software assumes the bdaddr is in the phys field
 2) there is no ioctl on the hidraw device to get the unique identifier

The second one is the killer, as we could update the software, but
we would then also have to add a new ioctl to the hidraw driver.

Source: SKY
Upstream-Status: Pending
Signed-off-by: Tijo Thomas John <tijo.thomas@sky.uk>
###################################################################################################

---
 profiles/input/hog-lib.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/profiles/input/hog-lib.c b/profiles/input/hog-lib.c
index dcf7f60..4adbd89 100644
--- a/profiles/input/hog-lib.c
+++ b/profiles/input/hog-lib.c
@@ -989,7 +989,7 @@ static void report_map_read_cb(guint8 status, const guint8 *pdu, guint16 plen,
 	ev.type = UHID_CREATE;
 
 	bt_io_get(g_attrib_get_channel(hog->attrib), &gerr,
-			BT_IO_OPT_SOURCE, ev.u.create.phys,
+			BT_IO_OPT_DEST, ev.u.create.phys,
 			BT_IO_OPT_DEST, ev.u.create.uniq,
 			BT_IO_OPT_INVALID);
 	if (gerr) {
-- 
2.18.0

